geode-core/src/integrationTest/java/org/apache/geode/internal/cache/PartitionedRegionIntegrationTest.java

```diff
@@ -49,10 +70,37 @@ public void bucketSorterShutdownAfterRegionDestroy() {
 
   @Test
   public void bucketSorterIsNotCreatedIfNoEviction() {
+    server.startServer();
     PartitionedRegion region =
         (PartitionedRegion) server.createRegion(RegionShortcut.PARTITION, "PR1",
             rf -> rf.setOffHeap(false));
     ScheduledExecutorService bucketSorter = region.getBucketSorter();
     assertThat(bucketSorter).isNull();
   }
+
+  @Test
+  // See GEODE-7106
+  public void generatePRIdShouldNotThrowNumberFormatExceptionIfAnErrorOccursWhileReleasingTheLock()
+      throws IOException {
+    String methodName = testName.getMethodName();
+    File customLogFile = temporaryFolder.newFile(methodName + ".log");
+    DistributedLockService mockLockService = mock(DistributedLockService.class);
+    doReturn(true).when(mockLockService).lock(any(), anyLong(), anyLong());
+    doThrow(new RuntimeException("Mock Exception")).when(mockLockService).unlock(any());
+
+    server.withProperty("log-level", "FINE")
+        .withProperty("log-file", customLogFile.getAbsolutePath()).startServer();
+    PartitionedRegion region =
+        spy((PartitionedRegion) server.createRegion(RegionShortcut.PARTITION, methodName));
+    doReturn(mockLockService).when(region).getPartitionedRegionLockService();
+
+    assertThatCode(() -> region.generatePRId(server.getCache().getInternalDistributedSystem()))
+        .doesNotThrowAnyException();
+    assertThat(Files.lines(customLogFile.toPath())
```

Comment:
Can this check be done by looking at logger.warn invocation? If there was any exception the logger.warn would not have successfully called. In that case we could remove checking for the exception string in the log file.

Reply:
Done!